Conjur/conventions/ruby-style-guide.md

```diff
@@ -24,10 +23,163 @@ This guide is intended to be a "living document" in the sense that it's expected
 as we refine and hone our Ruby knowledge base.
 
 ## Style Baseline
-- **Guide to Ruby** ([Why's Poignant Guide](http://poignant.guide/)) - The best place to start for
-  reading about the langauge
-- **bbatsov's style guide** ([Style Guide](https://github.com/bbatsov/ruby-style-guide)) - A
-  community-driven collection of best practices for Ruby development
+- Follow this official guidelines of the Ruby language https://docs.ruby-lang.org/en/master/doc/syntax/methods_rdoc.html
+  
+- Methods whose purpose is to produce a side effect (change a file, make a network call, raise an error, etc) should have verb phrase names. 
+
+  ```ruby
+  def call
+    validate_and_decode_token
+    get_jwt_identity_from_request
+    validate_host_has_access_to_webservice
+    validate_origin
+    validate_restrictions
+    audit_success
+    new_token
+  rescue => e
+    audit_failure(e)
+    raise e
+  end
+  ```
+
+  
+
+- Functions that return an object should have the name of the returned object so these functions should not have verb phrase names.
+
+- For low-level style decisions about formatting, etc, we follow the ruby style. We advice to use rubocop to auto correct offence in the files you change before submitting to PR.
+
+- Dont add "name stutter" to the names of files/classes/functions/members if its already in the name of the package because its obvious . For example inside authn_jwt package there is no need to add the "name stutter" jwt to whats inside because its obvious.
```

Comment:
Small update for correct terminology:

Avoid "name stutter".  That is, don't repeat a name from the module path in a class, and don't repeat a name from the class in a method.   For example:
```ruby
# bad
class MyCar
  def start_car_engine
    # ...
  end
end

# good
class MyCar
  def start_engine
    # ...
  end
end

# bad
module Authenticators
  class OktaAuthenticator
    #...
  end
end

# good
module Authenticators
  class Okta
    #...
  end
end